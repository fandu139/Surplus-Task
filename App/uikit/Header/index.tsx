import AppStyles from '../../theme/appStyles';
import Colors from '../../theme/colors';
import Fonts from '../../theme/fonts';
import Text from '../../uikit/Text';
import React from 'react';
import { View, StyleSheet, StyleProp, ViewStyle } from 'react-native';

/**
 *
 * @param leftButton
 * @param leftContent
 * @param title
 * @param subtitle
 * @param rightButton
 * @returns {*}
 * @constructor
 */

interface Props {
  leftButton?: React.ReactNode;
  leftContent?: React.ReactNode;
  title: string;
  titleSelectable?: boolean;
  subtitle?: string;
  subtitleSelectable?: boolean;
  titleStyle?: StyleProp<ViewStyle>;
  rightButton?: React.ReactNode;
  testIDTitle?: string;
  testIDSubtitle?: string;
  accessibilityLabelTitle?: string;
  accessibilityLabelSubtitle?: string;
  headerContainerStyle?: StyleProp<ViewStyle>;
}

const Header: React.FC<Props> = ({
  leftButton,
  leftContent,
  title,
  titleSelectable,
  subtitle,
  subtitleSelectable,
  titleStyle,
  rightButton,
  testIDTitle,
  testIDSubtitle,
  accessibilityLabelTitle,
  accessibilityLabelSubtitle,
  headerContainerStyle,
}: Props) => (
  <View style={[styles.header, headerContainerStyle]}>
    <View style={styles.leftContent}>
      {leftButton}
      {leftContent}
      {subtitle ? (
        <View>
          <Text.Black
            style={[styles.titleAlt, titleStyle]}
            testID={testIDTitle}
            accessibilityLabel={accessibilityLabelTitle || testIDTitle}
            selectable={titleSelectable}
          >
            {title}
          </Text.Black>
          <Text
            style={styles.subtitle}
            testID={testIDSubtitle}
            accessibilityLabel={accessibilityLabelSubtitle || testIDSubtitle}
            selectable={subtitleSelectable}
          >
            {subtitle}
          </Text>
        </View>
      ) : (
        <Text.Black
          style={[styles.title, titleStyle]}
          testID={testIDTitle}
          accessibilityLabel={accessibilityLabelTitle || testIDTitle}
          selectable={titleSelectable}
        >
          {title}
        </Text.Black>
      )}
    </View>
    {rightButton}
  </View>
);

const styles = StyleSheet.create({
  header: {
    ...AppStyles.rowItemsCenterSpace,
    width: '100%',
    height: 60,
    backgroundColor: Colors.GREEN_47,
    paddingHorizontal: 20,
  },
  leftContent: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  title: {
    marginTop: 1,
    color: Colors.WHITE,
    fontSize: Fonts.size.large,
  },
  titleAlt: {
    marginTop: 1,
    color: Colors.WHITE,
    fontSize: Fonts.size.small,
  },
  subtitle: {
    marginTop: 1,
    color: Colors.WHITE,
  },
});

Header.defaultProps = {
  leftButton: undefined,
  leftContent: undefined,
  subtitle: undefined,
  rightButton: undefined,
  testIDTitle: 'header-text-title',
  testIDSubtitle: 'header-text-subtitle',
  accessibilityLabelTitle: undefined,
  accessibilityLabelSubtitle: undefined,
  titleSelectable: false,
  subtitleSelectable: false,
  titleStyle: {},
  headerContainerStyle: {},
};

export default Header;
