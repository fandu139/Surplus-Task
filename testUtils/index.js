import AuthContext from '../App/context/Auth';
import React from 'react';
import { act, fireEvent, render } from '@testing-library/react-native';

const customRender = (ui, options) => {
  const isAuthenticated = options?.isAuthenticated || false;
  const renderOptions = options?.rendererOptions;
  const authContext = options?.authContext;
  return render(
    <AuthContext.Provider value={{ isAuthenticated, ...authContext }}>{ui}</AuthContext.Provider>,
    renderOptions,
  );
};

// override render method
export { customRender as render };

export function wait(ms) {
  return act(() => new Promise((r) => setTimeout(r, ms)));
}

// Due to debounce on Button UI Kit, we need to add milisecond waiting timer
export function fireAsyncPress(element, ms = 100) {
  fireEvent.press(element);
  return wait(ms);
}
