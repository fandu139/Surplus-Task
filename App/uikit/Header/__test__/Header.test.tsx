import Header from '../';
import React from 'react';
import { render } from '../../../../testUtils';

describe('Header component test', () => {
  describe('Header snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<Header title="Header Title" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Header with Subtitle snapshot test', () => {
    it('should render correctly', () => {
      const tree = render(<Header title="Header Title" subtitle="Header Subtitle" />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
