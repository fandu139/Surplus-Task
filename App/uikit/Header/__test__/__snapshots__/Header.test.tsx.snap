// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component test Header snapshot test should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#47bba5",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "width": "100%",
      },
      {},
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <Text
      accessibilityLabel="header-text-title"
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "#202020",
            "fontSize": 14,
            "fontWeight": "normal",
          },
          [
            {
              "color": "#FFFFFF",
              "fontSize": 18,
              "marginTop": 1,
            },
            {},
          ],
          undefined,
          undefined,
        ]
      }
      testID="header-text-title"
    >
      Header Title
    </Text>
  </View>
</View>
`;

exports[`Header component test Header with Subtitle snapshot test should render correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "backgroundColor": "#47bba5",
        "flexDirection": "row",
        "height": 60,
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "width": "100%",
      },
      {},
    ]
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
      }
    }
  >
    <View>
      <Text
        accessibilityLabel="header-text-title"
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "#202020",
              "fontSize": 14,
              "fontWeight": "normal",
            },
            [
              {
                "color": "#FFFFFF",
                "fontSize": 14,
                "marginTop": 1,
              },
              {},
            ],
            undefined,
            undefined,
          ]
        }
        testID="header-text-title"
      >
        Header Title
      </Text>
      <Text
        accessibilityLabel="header-text-subtitle"
        allowFontScaling={false}
        error={false}
        link={false}
        selectable={false}
        style={
          [
            {
              "color": "#202020",
              "fontSize": 14,
              "fontWeight": "normal",
            },
            {
              "color": "#FFFFFF",
              "marginTop": 1,
            },
            false,
            false,
          ]
        }
        testID="header-text-subtitle"
      >
        Header Subtitle
      </Text>
    </View>
  </View>
</View>
`;
